src/jvm/main/org/jetbrains/kotlinx/lincheck/TransformationClassLoader.java

```diff
@@ -49,18 +49,14 @@ public class TransformationClassLoader extends ExecutionClassLoader {
     private static final Map<String, byte[]> stressStrategyBytecodeCache = new ConcurrentHashMap<>();
     private static final Map<String, byte[]> modelCheckingStrategyBytecodeCache = new ConcurrentHashMap<>();
 
-    public TransformationClassLoader(Strategy strategy, Runner runner) {
+    public TransformationClassLoader(Strategy strategy) {
         classTransformers = new ArrayList<>();
         // Apply the strategy's transformer at first, then the runner's one.
-        if (strategy.needsTransformation()) classTransformers.add(strategy::createTransformer);
-        if (runner.needsTransformation()) classTransformers.add(runner::createTransformer);
-        remapper = UtilsKt.getRemapperByTransformers(
-                // create transformers just for their class information
-                classTransformers.stream()
-                        // pass any parameter to Transformer constructors - it won't be used
-                        .map(constructor -> constructor.apply(new TraceClassVisitor(null)))
-                        .collect(Collectors.toList())
-        );
+        // TODO: add Strategy.transformationMode property to get rig of this instanceof check
+        //  (now it's impossible due to lack of internal modifier in Java)
+        TransformationMode transformationMode = strategy instanceof ModelCheckingStrategy ? TransformationMode.MODEL_CHECKING : TransformationMode.STRESS;
+        classTransformers.add((cw) -> new LincheckClassVisitor(transformationMode, cw));
```

Comment:
It is always one transformer now, you don't need to use a list

Reply:
You also don't need this constructor:

`TransformationClassLoader(Function<ClassVisitor, ClassVisitor> classTransformer)`

Reply:
No, it's not true, as it needed in LTS:
<img width="1027" alt="image" src="https://github.com/JetBrains/lincheck/assets/45040429/ca4e8537-e002-4f2d-9d64-2441383d6216">